release/run_release_test.sh

```diff
@@ -69,6 +69,23 @@ if [ -z "${NO_CLONE}" ]; then
   echo "Cloning test repo ${RAY_TEST_REPO} branch ${RAY_TEST_BRANCH}"
   git clone --depth 1 -b "${RAY_TEST_BRANCH}" "${RAY_TEST_REPO}" "${TMPDIR}"
   pushd "${TMPDIR}/release" || true
+  HEAD_COMMIT=$(git rev-parse HEAD)
+  echo "The cloned test repo has head commit of ${HEAD_COMMIT}"
+
+  # We only do this if RAY_TEST_REPO and RAY_TEST_BRANCH are pointing to ray master.
+  # Theoretically, release manager may also run into this issue when manually triggering
+  # release test runs. But cherry-picks are rare and thus it's less likely to run into
+  # this racing condition, ignoring for now.
+  if [ "${RAY_TEST_REPO}" == "https://github.com/ray-project/ray.git" ] && \
+  [[ "${PARSED_RAY_WHEELS}" == *"master"*  ]] && \
+  [ "${RAY_TEST_BRANCH}" == "master" ] && [ "${BUILDKITE_COMMIT}" ] && \
+  [ "${HEAD_COMMIT}" != "${BUILDKITE_COMMIT}" ]; then
+    echo "The checked out test code doesn't match with the installed wheel. \
+This is likely due to a racing condition when a PR is landed between \
+a wheel is installed and test code is checked out."
+    echo "Hard resetting from ${HEAD_COMMIT} to ${BUILDKITE_COMMIT}."
+    git reset --hard "${BUILDKITE_COMMIT}"
```

Comment:
I think this "might" fail because we use git clone --depth 1

Reply:
good catch. Thanks! wasn't aware of the difference of `--depth`: Create a shallow clone with a history truncated to the specified number of commits.

How about let's just remove it? I did a quick search of how to clone to a specific commit sha - it doesn't seem well supported.

Reply:
I think removing it is fine yes